internal/config/config.go

```diff
@@ -0,0 +1,249 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+// RepositoryConfig represents a repository configuration
+type RepositoryConfig struct {
+	Name         string   `yaml:"name"`
+	ExcludeDirs  []string `yaml:"exclude_dirs"`
+	ExcludeFiles []string `yaml:"exclude_files"`
+	Owners       []string `yaml:"-"` // Not serialized, used for CODEOWNERS
+}
+
+// Writer writes repository configurations to disk
+type Writer struct {
+	reposDir       string
+	codeownersFile string
+}
+
+// NewWriter creates a new configuration writer
+func NewWriter(reposDir, codeownersFile string) *Writer {
+	return &Writer{
+		reposDir:       reposDir,
+		codeownersFile: codeownersFile,
+	}
+}
+
+// Write writes a repository configuration to disk
+func (w *Writer) Write(cfg RepositoryConfig, dryRun bool) error {
+	// Validate repository name
+	if strings.TrimSpace(cfg.Name) == "" {
+		return fmt.Errorf("repository name cannot be empty")
+	}
+	// Allow "org/repo" or "repo" only
+	if strings.Count(cfg.Name, "/") > 1 {
+		return fmt.Errorf("invalid repository name format: %q", cfg.Name)
+	}
+	// Disallow path traversal characters
+	if strings.Contains(cfg.Name, "..") || strings.ContainsAny(cfg.Name, `\`) {
+		return fmt.Errorf("invalid characters in repository name: %q", cfg.Name)
+	}
+
+	// Generate filename from repository name
+	filename := w.getFilename(cfg.Name)
+
+	var targetPath string
+	if dryRun {
+		dryRunDir := filepath.Join(filepath.Dir(w.reposDir), "discovered-repos")
+		targetPath = filepath.Join(dryRunDir, filename)
+	} else {
+		targetPath = filepath.Join(w.reposDir, filename)
+	}
+
+	data, err := yaml.Marshal(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to marshal config: %w", err)
+	}
+
+	// Write atomically
+	if err := atomicWrite(targetPath, data, "."+filename+".tmp.*"); err != nil {
+		return fmt.Errorf("failed to write config to %s: %w", targetPath, err)
+	}
+
+	// Update CODEOWNERS if not dry run
+	if !dryRun {
+		if err := w.updateCodeowners(filename, cfg.Owners); err != nil {
+			return fmt.Errorf("failed to update CODEOWNERS: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// getFilename generates a filename from repository name
+func (w *Writer) getFilename(repoName string) string {
+	// Extract repo name from "org/repo" format
+	parts := strings.Split(repoName, "/")
+	if len(parts) == 2 {
+		return parts[1] + ".yaml"
+	}
+	return repoName + ".yaml"
+}
+
+// atomicWrite writes data to targetPath atomically using temp file + rename
+func atomicWrite(targetPath string, data []byte, tempPattern string) error {
```

Comment:
Why are atomic writes needed? From what I can tell, there's only a single process writing to these files sequentially.